DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -414,6 +428,84 @@ public Task ModifyPositionAsync(int position, string reason = null)
             return this.Discord.ApiClient.ModifyGuildChannelPositionAsync(this.Guild.Id, pmds, reason);
         }
 
+        /// <summary>
+        /// Updates the channel parent, moving the channel to the bottom of the new category.
+        /// </summary>
+        /// <param name="parent_id">New parent ID. Will move out of parent if null.</param>
+        /// <param name="lock_permissions">Sync permissions with parent. Defaults to null.</param>
+        /// <param name="reason">Reason for audit logs.</param>
+        /// <returns></returns>
+        /// <exception cref="Exceptions.UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ManageChannels"/> permission.</exception>
+        /// <exception cref="Exceptions.NotFoundException">Thrown when the channel does not exist.</exception>
+        /// <exception cref="Exceptions.BadRequestException">Thrown when an invalid parameter was provided.</exception>
+        /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+        public Task ModifyParentAsync(ulong? parent_id = null, bool? lock_permissions = null, string reason = null)
+        {
+            if (this.Guild == null)
+                throw new InvalidOperationException("Cannot modify parent of non-guild channels.");
+            if (this.IsCategory)
+                throw new InvalidOperationException("Cannot modify parent of category channels.");
```

Comment:
throw ArgumentException instead

Reply:
Also going to change exception thrown in `ModifyPositonAsync()` to match. Exception signature is a breaking change, but this PR I imagine already has a few breaking changes. Can revert that change easily though.

Reply:
Done.